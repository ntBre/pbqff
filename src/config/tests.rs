use insta::{assert_debug_snapshot, with_settings};
use test_case::test_case;

use super::*;

#[test_case("testfiles/test.toml" ; "cart")]
#[test_case("testfiles/path.toml" ; "path templates")]
#[test_case("testfiles/normal.toml" ; "norm")]
fn load_config(path: &str) {
    with_settings!({ snapshot_suffix => path }, {
        assert_debug_snapshot!(Config::load(path));
    });
}
