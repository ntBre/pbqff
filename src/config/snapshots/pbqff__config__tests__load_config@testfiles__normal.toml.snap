---
source: src/config/tests.rs
expression: "Config::load(path)"
snapshot_kind: text
---
Config {
    geometry: Zmat(
        "O\nH 1 OH\nH 1 OH 2 HOH\n\nOH = 1.0\nHOH = 109.5\n",
    ),
    optimize: true,
    charge: 0,
    step_size: 0.005,
    coord_type: Normal,
    template: "scfcrt=1.D-21 aux(precision=14 comp xp xs xw) PM6 THREADS=1",
    hybrid_template: "scfcrt=1.D-21 aux(precision=14 comp xp xs xw) PM6 THREADS=1",
    queue_template: None,
    program: Mopac,
    queue: Local,
    sleep_int: 2,
    job_limit: 2048,
    chunk_size: 1,
    findiff: false,
    check_int: 0,
    weights: None,
    dummy_atoms: None,
    norm_resume_hff: true,
}
